test/integration-tests/Seaport.test.ts

```diff
@@ -252,14 +253,81 @@ describe('Seaport', () => {
   })
 })
 
-describe('SeaportV1_4', () => {
+describe('Seaport v1.4', () => {
   let alice: SignerWithAddress
+  let bob: SignerWithAddress
   let router: UniversalRouter
   let permit2: Permit2
   let planner: RoutePlanner
   let cryptoCovens: ERC721
+  let galaContract: Contract
+  let townStarNFT: Contract
 
-  describe('SeaportV1_4 ETH -> NFT', () => {
+  describe('ERC20 -> NFT', () => {
+    beforeEach(async () => {
+      await resetFork(16784348 - 1) // 1 block before the order was created
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [ALICE_ADDRESS],
+      })
+      alice = await ethers.getSigner(ALICE_ADDRESS)
+      galaContract = new ethers.Contract(GALA.address, TOKEN_ABI, alice)
+
+      // alice can't sign permits as we don't have her private key. Instead bob is used
+      bob = (await ethers.getSigners())[1]
+      permit2 = (await deployPermit2()).connect(bob) as Permit2
+      router = (await deployUniversalRouter(permit2)).connect(bob) as UniversalRouter
+      planner = new RoutePlanner()
+      townStarNFT = TOWNSTAR_1155.connect(bob) as ERC1155
+      await galaContract.connect(bob).approve(permit2.address, ethers.constants.MaxUint256)
+
+      // Alice seeds bob's account with GALA for tests
+      await galaContract.transfer(bob.address, 100000 * 10 ** 8)
+    })
+
+    it('completes a fulfillAdvancedOrder type', async () => {
+      let { advancedOrder, value } = getAdvancedOrderParams(seaportV1_4Orders[1])
+      value = value.div(2) // the numerator/denominator mean this is a partial fill
+      const params = advancedOrder.parameters
+      const considerationToken = params.consideration[0].token
+      const tokenId = params.offer[0].identifierOrCriteria
+
+      const calldata = seaportV1_4Interface.encodeFunctionData('fulfillAdvancedOrder', [
+        advancedOrder,
+        [],
+        OPENSEA_CONDUIT_KEY,
+        bob.address,
+      ])
+
+      const permit = {
+        details: {
+          token: considerationToken,
+          amount: value,
+          expiration: 0, // expiration of 0 is block.timestamp
+          nonce: 0, // this is his first trade
+        },
+        spender: router.address,
+        sigDeadline: DEADLINE,
+      }
+      const sig = await getPermitSignature(permit, bob, permit2)
+
+      planner.addCommand(CommandType.APPROVE_ERC20, [considerationToken, 0])
+      planner.addCommand(CommandType.PERMIT2_PERMIT, [permit, sig])
+      planner.addCommand(CommandType.PERMIT2_TRANSFER_FROM, [considerationToken, router.address, value])
+      planner.addCommand(CommandType.SEAPORT_V1_4, [0, calldata])
+
+      const { commands, inputs } = planner
+      await expect(await townStarNFT.balanceOf(bob.address, tokenId)).to.eq(0)
+      await expect(router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE)).to.changeTokenBalance(
```

Comment:
waaiiit what is this??? lol `changeTokenBalance` kinda cool

Reply:
its very cool .. alice showed me it too!